classes/module/CarrierModule.php

```diff
@@ -23,10 +23,40 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
+
+/**
+ * Base class to be used instead of `Module` for modules that manage their own carriers.
+ *
+ * A module can declare one or several carriers. For each carrier managed by the module:
+ *   - The carrier's `shipping_external` flag must be set to `true`.
+ *   - The carrier's `external_module_name` must match the module's technical name.
+ *
+ * This ensures that PrestaShop delegates shipping-cost calculation to the module
+ * through the `CarrierModuleCore` methods.
+ */
 abstract class CarrierModuleCore extends Module
 {
+    /**
+     * Calculate the shipping cost for an order.
+     *
+     * @param CartCore $params the cart instance
+     * @param float $shipping_cost the base shipping cost before module adjustments
+     *
+     * @return float|false|null Returns a float for shipping cost.
+     *                          Returns null to mark the delivery options in the checkout tunnel as "free".
+     *                          Returns false to filter out (exclude) this delivery option.
+     */
     abstract public function getOrderShippingCost($params, $shipping_cost);
 
+    /**
+     * Calculate the external shipping cost for an order. Called when `need_range` of the carrier is set as `false`.
+     *
+     * @param CartCore $params the cart instance
```

Comment:
Cart was actually better, we usually don't suffix with `Core`

```suggestion
     * @param Cart $params the cart instance
```

Reply:
That's what I did initially, but PHPstan doesn't accept it. In the CartCore class, we call these functions with an instance of the class (https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Cart.php#L4002), so it will have an impact. Furthermore, I see that quite a few places in the legacy classes use the "MyClassCore" type.

- https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Guest.php#L250
- https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Carrier.php#L710
- https://github.com/PrestaShop/PrestaShop/blob/develop/classes/CartRule.php#L1104
...